### PR TITLE
Switch to JDK 20

### PR DESCRIPTION
**Pull request by koppor**
Follow-up to https://github.com/JabRef/jabref/pull/9766.

I updated the Java version.

I also added a [markdown linter](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) and a [spell checker](https://marketplace.visualstudio.com/items?itemName=valentjn.vscode-ltex) to the recommended VS Code extensions.

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
can we try to use the toolchain stuff again?

**Comment by Siedlerchr** [comment]
Seems like modernizer fails for some reason

**Comment by koppor** [comment]
Toolchain refs https://github.com/JabRef/jabref/pull/9695

**Comment by koppor** [comment]
Modernizer error:

```
 > Task :modernizer FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':modernizer'.
> java.lang.reflect.InvocationTargetException
```

**Comment by koppor** [comment]
Modernizer fixes by updating to most recent version.

Can the toolchain thing posponed or do you see some crucial benefit?

File: .github/workflows/deployment-arm64.yml [hunk 1]
```diff
@@ -59,8 +59,8 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
-          java-version: 19
-          distribution: 'liberica'
+          java-version: 20
+          distribution: 'temurin'
```
Comment (calixtus) [inline .github/workflows/deployment-arm64.yml:63]: @Siedlerchr please confirm

Comment (Siedlerchr) [reply .github/workflows/deployment-arm64.yml:63]: should be fine Using temurin for development as well. I think it was leftover from the last time I experimented with different jdks for signing.

**Comment by koppor** [comment]
In [`dfd5f51` (#9767)](https://github.com/JabRef/jabref/pull/9767/commits/dfd5f510e7337f1c10f46a553a93387e27671323), I re-enabled creation of the portable app. Just as "workaround" as long as notarization is disabled. I think, it's good to have a consistent build - and work on the Mac OS notarization in a follow-up PR.

**Review by calixtus — APPROVED** [review]

**Pull request merged**